### PR TITLE
Add Tests and some crucial changes

### PR DESCRIPTION
**Pull request by yuvals2146**
What in this large pr:
1. change the .env files to .envrc to support direnv
2. add direnv (https://direnv.net/) 
3. add running scripts in bin folder:
   a. test
   b. run
   c. logger (helper)
4. add db commends in package json to help with prisma cmds
5. add cascade delete to positionInfo table
6. cleaning the few savePositionDataSQL remains 
7. add some try/catch and fix alerts to getPositionData
8. add condition to NOT log if in test for readable test results 
9. create mock file for test objects
10. create factory file for test help functions
11. add yarn.lock

Few things: 
1. form now on we should use yarn instead of npm 

> The major difference between NPM and Yarn comes in terms of security performance. While NPM installs packages sequentially, Yarn performs parallel installation resulting in better speed and performance. NPM has tried to fix vulnerabilities, but still, Yarn is considered more secure than NPM.

2. to run our code lets start use `. bin/start` and `.bin/test/` the bash script can hold more logic and conditions
3. form now on lets try to make our process test driven development and start from the test other wise we might neglect this part in our 'move ultra fast and break everything'

Let me know if anything is unclear (please do not hesitate to ask me for rename,change,rewrite or anything. everything here is open for changes ❤️ 🚀 🌔 )

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -1,5 +1,5 @@
 node_modules
+
 # Keep environment variables out of version control
-.env
-.env.test
-.wwebjs_cache/*
```
Comment (azrield) [inline .gitignore:5]: what's happened to wwebjs_cache?

Comment (yuvals2146) [reply .gitignore:5]: we are not using whatsapp-web package anymore so it is not relavnt

File: envrc.example [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+
+#notifer settings:
+export UNISWAP_POOL_URL_PREFIX=""
+export UNISWAP_POSITION_URL_PREFIX=""
+export WHATSAPP_USER_DANY=""
+export WHATSAPP_USER_YUVAL=""
+export WHATSAPP_ALERT_GROUP=""
+
+#logger settings:
+
```
Comment (azrield) [inline envrc.example:32]: #notifer settings:
PUSHOVER_TOKEN = 'abscf1pn7i2x19cehxpbaqyyc47uf4'
PUSHOVER_YUVAL = 'uffrjpkze2v9tmar3vgbzdhed4p3i7'
PUSHOVER_DANY =  'u1wqegs8wyw4s71gvg8n4591267ote'

File: envrc.example [hunk 3]
```diff
@@ -0,0 +1,35 @@
...
+# only for envrc.test!
+
+export SUITE_TO_RUN=""
```
Comment (azrield) [inline envrc.example:35]: #don't repeat last alert unless an hour passed
REPEAT_ALERT_INTERVAL = 60000000

File: src/blockchain/getPostionData.js [hunk 4]
```diff
@@ -252,8 +254,29 @@ async function getFees(
   return fees;
 }
 
+// do something with me!
```
Comment (azrield) [inline src/blockchain/getPostionData.js:257]: remove

File: src/blockchain/getPostionData.js [hunk 5]
```diff
@@ -534,23 +568,28 @@ const retriveInitalPositionData = async (position, txHash = null) => {
...
+      throw new Error(
+        `No inital data found for position ${position.id} reason ${e.message}`
+      );
     } else {
-      logger.error(e.message);
-      logger.error("No init data found for position", position.id);
       notify(
         "🪙🕰️ Action Needed 🕰️🪙",
-        `No initial data found for position ${position.id} on chain ${
+        `No historical data found for position ${position.id} on chain ${
```
Comment (azrield) [inline src/blockchain/getPostionData.js:584]: same - add chain name

File: src/test/blockchain/getPostionData.test.js [hunk 6]
```diff
@@ -0,0 +1,238 @@
...
+  retriveInitalPositionData,
+} = require("../../blockchain/getPostionData");
+const {
+  mockEthereumPositionOneItitialData,
+  mockArbitPositionOneItitialData,
+} = require("../mocks.js");
+const { chains, chainsNames } = require("../../utils/chains");
+// async function getData(position) {...} return PositionInfo;
+
+const mockPosition = {
```
Comment (azrield) [inline src/test/blockchain/getPostionData.test.js:13]: remove?

File: src/test/blockchain/getPostionData.test.js [hunk 7]
```diff
@@ -0,0 +1,238 @@
...
+  getPostionData,
+  getCurrentBlockNumber,
+  retriveInitalPositionData,
+} = require("../../blockchain/getPostionData");
+const {
+  mockEthereumPositionOneItitialData,
+  mockArbitPositionOneItitialData,
+} = require("../mocks.js");
+const { chains, chainsNames } = require("../../utils/chains");
+// async function getData(position) {...} return PositionInfo;
```
Comment (azrield) [inline src/test/blockchain/getPostionData.test.js:11]: remove comment

File: src/test/blockchain/getPostionData.test.js [hunk 8]
```diff
@@ -0,0 +1,238 @@
...
+  etherUsdExchangeRate: 1,
+  ArbitUsdExchangeRate: 1,
+  positionId: 1,
+  blockNumber: 2,
+};
+const mockPositionChain = 1;
+describe("getPostionData", () => {
+  test("should get position data for valid etherum position", async () => {
+    const resultData = await getPostionData({
+      id: 482139,
```
Comment (azrield) [inline src/test/blockchain/getPostionData.test.js:39]: this is problematic - this position will die somewhen

File: src/test/blockchain/getPostionData.test.js [hunk 9]
```diff
@@ -0,0 +1,238 @@
...
+  test("should not get position data for not exist token", async () => {
+    expect(async () => {
+      await getPostionData({ id: 10000000000, chain: 1 });
+    }).rejects.toThrow(
+      /could not get data for position 10000000000 on chain etherum/
+    );
+  });
+});
+
+// ON HOLD UNTIL DANY FINISH THIS PART
```
Comment (azrield) [inline src/test/blockchain/getPostionData.test.js:108]: uncomment

File: src/test/blockchain/getPostionData.test.js [hunk 10]
```diff
@@ -0,0 +1,238 @@
...
+//const getPoolExchangeRate = async (poolAddress, chain) => {...} return exchangeRate;
+// describe("getPoolExchangeRate", () => {
+//   test("should get Exchange rate from arbitrum for a valid token0?? pool address and chain", async () => {
+//     //     getPoolExchangeRate(validPoolsAddress.arb0, cha);
+//     expect(1).toEqual(1);
+//   });
+// });
+// ON HOLD UNTIL DANY FINISH THIS PART
+
+//const getCurrentBlockNumber = async (chain) => {...} return blockNumber;
```
Comment (azrield) [inline src/test/blockchain/getPostionData.test.js:118]: remove?

File: src/test/blockchain/getPostionData.test.js [hunk 11]
```diff
@@ -0,0 +1,238 @@
...
+      {
+        id: 482139,
+        chain: chainsNames.etherum,
+      },
+      txHash
+    );
+    expect(resultData).toEqual(mockEthereumPositionOneItitialData);
+  });
+
+  // no support for arbit subgraph for now
```
Comment (azrield) [inline src/test/blockchain/getPostionData.test.js:163]: remove comment?

File: src/test/mocks.js [hunk 12]
```diff
@@ -0,0 +1,225 @@
...
+  createdAt: 1680613919000,
+  initValueToken0: 0.000000000965856802,
+  token0Symbol: "USDC",
+  initValueToken1: 1.999999998264815215,
+  token1Symbol: "ETH",
+  initToken0USDRate: 0.9995,
+  initToken1USDRate: 1872.32,
+};
+
+const mockEthereumPositionOneItitialData = {
```
Comment (azrield) [inline src/test/mocks.js:24]: Ititial=Initial

File: src/test/mocks.js [hunk 13]
```diff
@@ -0,0 +1,225 @@
...
+  createdAt: 1693071326000,
+  initValueToken0: 0.329534836654748873,
+  token0Symbol: "ETH",
+  initValueToken1: 566.671305985847971808,
+  token1Symbol: "ARB",
+  initToken0USDRate: 1645.12,
+  initToken1USDRate: 0.9415,
+};
+
+const mockArbitPositionOneItitialData = {
```
Comment (azrield) [inline src/test/mocks.js:149]: Ititial=Initial

File: src/test/factories.js [hunk 14]
```diff
@@ -0,0 +1,42 @@
+require("dotenv").config({ path: `${__dirname}/../../.env.test` }); //initialize dotenv
```
Comment (azrield) [inline src/test/factories.js:1]: remove?

File: src/utils/queryTheGraph.js [hunk 15]
```diff
@@ -1,6 +1,6 @@
 const axios = require("axios");
-const logger = require("./logger.js");
-const { chains } = require("./chains.js");
+const logger = require("../utils/logger");
```
Comment (azrield) [inline src/utils/queryTheGraph.js:2]: require("logger");

File: src/utils/queryTheGraph.js [hunk 16]
```diff
@@ -1,6 +1,6 @@
...
+const { chains } = require("../utils/chains.js");
```
Comment (azrield) [inline src/utils/queryTheGraph.js:3]: same

**Pull request merged**